sdk/tables/Azure.Data.Tables/tests/TableServiceClientLiveTests.cs

```diff
@@ -228,6 +228,13 @@ public async Task GetTablesReturnsTablesWithFilter()
 
                 Assert.That(() => tableResponses, Is.Not.Empty);
                 Assert.That(() => tableResponses.Select(r => r.TableName), Contains.Item(tableName));
+
+                // Query with a filter.
+
+                tableResponses = (await service.GetTablesAsync(filter: t => t.TableName == tableName).ToEnumerableAsync().ConfigureAwait(false)).ToList();
+
+                Assert.That(() => tableResponses, Is.Not.Empty);
+                Assert.That(() => tableResponses.Select(r => r.TableName), Contains.Item(tableName));
```

Comment:
Won't this condition be true if the filter is not applied at all and we get all tables back?

Reply:
good catch - fixed.